packages/wrangler/src/pages/functions.tsx

```diff
@@ -0,0 +1,86 @@
+import { existsSync, readFileSync, writeFileSync } from "node:fs";
+import { FatalError } from "../errors";
+import {
+	isRoutesJSONSpec,
+	optimizeRoutesJSONSpec,
+} from "./functions/routes-transformation";
+import { pagesBetaWarning } from "./utils";
+import type { ArgumentsCamelCase, Argv } from "yargs";
+
+type OptimizeRoutesArgs = {
+	"routes-path": string;
+	"output-routes-path": string;
+};
+
+export function OptimizeRoutesOptions(yargs: Argv): Argv<OptimizeRoutesArgs> {
+	return yargs
+		.options({
+			"routes-path": {
+				type: "string",
+				demandOption: true,
+				description: "The location of the _routes.json file",
+			},
+		})
+		.options({
+			"output-routes-path": {
```

Comment:
And then this one optional, such that if omitted, we overwrite the input file. Happy to be overruled on this though.